### ISSUE TITLE
Lua weirdness

### ISSUE THREAD

**Issue by raingloom** [codecombat/codecombat#3362]
My sister tried the following:
```
- If there is an enemy, attack it.
-- Otherwise, attack the chest!
while true do
    self:say(_VERSION)
    -- Use if/else.
    self:attack("Chest" ) 
    if enemy=self:findNearestEnemy() then
        self:attack (enemy)

end
end
```
This should give a huge compilation error. because of the `if a=b then...` part.
In Lua `a=b` is a statement, not an expression. I'm not sure what implementation the game runs, but repl.it's gives a clear error.

[repl.it example](https://repl.it/BhVf)

**Comment by basicer** [comment]
Good catch. The game runs my custom lua implementation that compiles to a javascript AST.  It considers an assignment to be an expression https://github.com/basicer/lua2js/blob/master/lua.pegjs#L350  .  I'll look into changing it to a statement.
